test/tests/functional/pbs_reservations.py

```diff
@@ -1123,3 +1123,128 @@ def test_reservations_with_expired_subjobs(self):
         # Check status of the parent-job using qstat -fx once reservation ends
         self.server.expect(JOB, {'job_state': 'F', 'substate': '91'},
                            extend='x', id=jid)
+
+    def test_ASAP_resv_with_multinode_job_array(self):
+        """
+        Test multinode job array with ASAP reservation
+        """
+        if (len(self.moms) < 2):
+            self.skip_test("Test requires 2 moms: use -p mom1:mom2")
+        a = {'resources_available.ncpus': 4}
+        for mom in self.moms.values():
+            self.server.manager(MGR_CMD_SET, NODE, a, id=mom.shortname)
+        self.server.manager(MGR_CMD_SET, SERVER,
+                            {'job_history_enable': 'True'})
+        attrs = {ATTR_J: '1-5',
+                 'Resource_List.select': '2:ncpus=1',
+                 'Resource_List.walltime': '30',
+                 'Resource_List.place': 'scatter'}
+        j = Job(PBSROOT_USER)
+        j.set_sleep_time(30)
```

Comment:
Please consider reducing the sleeptime and walltime to something smaller (like 5 or 10 seconds) 30 seconds is too long